### DISCUSSION TITLE
Access to tabular values from FAO56 tables

### DISCUSSION THREAD

**Discussion by kthorp** [kthorp/pyfao56#55] [Ideas]
Several users have proposed incorporating data from FAO56 tables to enable automatic specification of input parameters. This likely could be enabled with functions and data added to the Parameters class.

**Comment by tpokoski** [comment]
Could do this by adding a class object that is essentially just a bunch of dictionaries for each table. Or a small database with all of the tables, and then a series of queries to put the data into Parameters classes. 

How would you imagine the automatic part? User would need to input their crop, and that's what queries for the parameters?

**Comment by kthorp** [comment]
Yes I think it would be a set of dictionaries in Parameters.  Perhaps a user instantiates Parameters with a crop string (i.e., "maize") and the parameters are set automatically.  I hesitate with some of this as pyfao56 should not be making claims about crop coefficients without strong support from literature.  For example, in Table 11 in FAO-56, the lengths of development stages vary not only by crop but also by region. Also, some of the crop coefficient values (Table 12, 17) are given as ranges, so what to do there.

**Comment by reaganames** [comment]
Hi Kelly! 

My name is Reagan Ames, and I am a student working with Tyler on updating the model. I am super excited to be working with you all on this! 

I just added a pull request addressing this discussion post, so feel free to take a look when you get a chance. 

Thanks!

**Reply by kthorp** [reply]
Thanks for getting this started.  I made some comments on the PR.  Happy to discuss over Zoom in needed.

**Reply by reaganames** [reply]
I've made some of the changes we discussed including adding a synonyms table, shifting the code into a new function within the parameters class, and adding user input checks. These changes are all fully functional and ready to go! Seeing how the branch has already been merged, what method would be easiest for you to access these changes? 

Thanks so much!

**Reply by kthorp** [reply]
Is it possible for you to submit another PR?

**Reply by reaganames** [reply]
Yes, added a pull request

**Comment by kthorp** [comment]
This topic is further discussed in PR #65 and #68. Many thanks to Reagan and Tyler for initial efforts.  I have now pushed a final implementation to the develop branch, and it will be released with v1.4.
